app/models/manageiq/providers/amazon/agent_coordinator.rb

```diff
@@ -214,6 +230,20 @@ def docker_auth
     @ems.authentications.find_by(:authtype => "smartstate_docker")
   end
 
+  # To run SSA, VPC needs to turn on enableDnsSupport and enableDnsHostnames
+  def get_dns_enabled_vpc_ids
+    vpc_ids = ec2.client.describe_vpcs.vpcs.map(&:vpc_id)
```

Comment:
Why `map` and `select`?  You should be able to:
```ruby
ec2.client.describe_vpcs.vpcs.select do |vpc|
  id = vpc.vpc_id
  enabled_dns_support?(id) && enabled_dns_hostnames?(id)
end
```

Reply:
Changed.